KK_ButtPlugin/ButtplugWsClient.cs

```diff
@@ -30,15 +33,18 @@ public void Open()
 
         public void Close()
         {
+            Devices.Clear();
+
             ButtPlugin.Logger.LogDebug("Disconnecting from Buttplug server.");
             websocket.Close();
             websocket.Dispose();
+            IsConnected = false;
         }
 
         public void LinearCmd(double position, int durationMs)
         {
             var commands = (
-                from device in devices
+                from device in Devices
                 where device.DeviceMessages.LinearCmd != null

```

Comment:
I know this is out of scope but you could use the IsStroker you just added here. Same for VibrateCmd.